test/cql-pytest/test_group_by.py

```diff
@@ -232,5 +232,37 @@ def test_group_by_non_aggregated_mutation_attribute_of_column(cql, table1):
     assert [row[1] for row in results] == [100000001, 100000003, 100000005, 100000007]
     assert all([[row[2] > 900000] for row in results])
 
+# This test is from cassandra_tests/validation/operations/select_group_by_test.py::testGroupByWithPaging()
+# Reproduces #21267
+@pytest.mark.xfail(reason="issue #21267")
+def test_group_by_static_column(cql, test_keyspace):
+
+    with new_test_table(cql, test_keyspace, "a int, b int, c int, s int static, d int, primary key (a, b, c)") as table:
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 1;')
+        cql.execute(f'UPDATE {table} SET s = 2 WHERE a = 2;')
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 3;')
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 4;')
```

Comment:
Looks good, but personally I would have tried to simplify the test to leave only the minimum required to show the bug. You probably don't need 4 different partitions in this test and perhaps 3 or even 2 would have been enough.